store/src/java/com/zimbra/cs/lmtpserver/ExternalEmailWarning.java

```diff
@@ -0,0 +1,200 @@
+/*
+ * ***** BEGIN LICENSE BLOCK *****
+ * Zimbra Collaboration Suite Server
+ * Copyright (C) 2022 Synacor, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify it under
+ * the terms of the GNU General Public License as published by the Free Software Foundation,
+ * version 2 of the License.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ * You should have received a copy of the GNU General Public License along with this program.
+ * If not, see <https://www.gnu.org/licenses/>.
+ * ***** END LICENSE BLOCK *****
+ */
+package com.zimbra.cs.lmtpserver;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.apache.commons.lang.StringEscapeUtils;
+
+import com.zimbra.common.localconfig.LC;
+
+/**
+ * ExternalEmailWarning is a singleton class that provides assorted utilities
+ * for the External Email Warning functionality.
+ * 
+ * @author telus
+ *
+ */
+public class ExternalEmailWarning {
+
+    private static ExternalEmailWarning instance;
+    private boolean isEnabled;
+    private String baseAdmonition;
+    private String textPlainAdmonition;
+    private String textHtmlAdmonition;
+
+    private ExternalEmailWarning() {
+        isEnabled = LC.zimbra_external_email_warning_enabled.booleanValue();
+        baseAdmonition = LC.zimbra_external_email_warning_message.value();
+        textPlainAdmonition = StringEscapeUtils.escapeHtml(baseAdmonition) + "\r\n";
+        textHtmlAdmonition = "<div><p>" + baseAdmonition + "</p><hr/></div>";
+    }
+
+    public static ExternalEmailWarning getInstance() {
+        if (instance == null) {
+            synchronized (ExternalEmailWarning.class) {
+                if (instance == null) {
+                    instance = new ExternalEmailWarning();
+                }
+            }
+        }
+        return instance;
+    }
+
+    /**
+     * Get whether External Email Warning feature is enabled.
+     * 
+     * @return true if EEW feature is enabled.
+     */
+    public boolean isEnabled() {
+        return isEnabled;
+    }
+
+    /**
+     * Get configured admonition or warning to be used on text/html mime message
+     * parts.
+     * 
+     * @return string with admonition or warning.
+     */
+    public String getTextHtmlAdmonition() {
```

Comment:
"Admonition" has the connotation of a reprimand, and we aren't _blaming_ the user for anything, but simply _warning_ them about a possible danger. It'd make more sense to s/admonition/warning throughout.